web/src/message_events.js

```diff
@@ -660,6 +660,8 @@ export function update_messages(events) {
                 //       with data fetched from the server (which is already updated)
                 //       when we move to new narrow and what data is locally available.
                 if (changed_narrow) {
+                    // Remove outdated cached data to avoid repopulating from it.
+                    message_list_data_cache.remove(new_filter);
```

Comment:
Can you expand the comment here with more reasoning on why this is correct or under what circumstances we could change the logic? In theory, we're changing the narrow because the messages moved, but if we were able to live-update the `MessageListData` for a cached destination narrow for the move, it should be fine. Is the issue that we've not attempted to do such a live-update of the `MessageListData` objects yet in the sequence in this file?

Reply:
Yeah, the main issue is we have't updated the cached data yet. Added a comment for that.